Dev

KlochkoVitaliy opened the pull request.

ZhosanDenis committed:
Изменения №1 от 05.04.2023
группа 1:
- добавлены зависимости в pom.xml;
- подключен liquibase, создан change log, создан файл со скриптами/SQL-запросами, созданы запросы на создание таблиц БД;
- созданы модели/сущности опекунов, собак, соглашений(договоров), ежедневных отчетов. Таблицы и сущность команд для бота (который через /) решил не создавать, т.к. делать запросы в БД просто для получения команды для сравнения с входящим сообщением нерационально - считаю, лучше хранить данные команды в коллекции и через метод искать и возвращать нужную.
- добавил пакеты и классы configuration и listener из шаблона для телеграмбота из предыдущей курсовой;
- добавил аннотацию @EnableScheduling над запускающим классом (Application), внес изменение в имя запускающего класса для соблюдения camelCase (AnimalShelterTelegrambotApplication -> AnimalShelterTelegramBotApplication)
группа 2:
- удалена таблица БД dogs, удален класс Dog, удалено поле dog в классе Carer;
- удалена таблица БД agreements, удален класс Agreement, удалено поле agreement в классе Carer;
- изменен тип данных колонки phone_number таблицы БД carers;
- удалены колонки passport_number, dog_id, agreement_id таблицы БД carers, удалены соответствующие данным колонкам поля класса Carer;
- в таблицу БД daily_reports добавлены колонки report_date, dog_diet, dog_health, dog_behavior, и соответствующие им поля (а также геттеры и сеттеры) в класс DailyReport.
группа 3:
- добавлен интерфейс CarerRepository для взаимодействия с БД;
- добавлен класс CarerService для добавления опекуна в БД;
- в TelegramBotConfiguration добавлен список команд (SetMyCommands);
- в TelegramBotUpdatesListener добавлены константы команд, заинжектен CarerService;
- код метода process() обернут в try-catch;
- добавлены обработки команд в зависимости от присланного сообщения или нажатой кнопки InlineKeyboard;
- добавлены методы для создания клавиатур основного меню и меню с информацией о приюте.
группа 4:
- добавлен комментарий к пути файла, отправляемого в ответ на запрос расписания работы приюта.
группа 5:
- добавлена @OpenAPIDefinition в AnimalShelterTelegramBotApplication для подключения Swagger UI;
- изменено название класса AnimalShelterTelegramBotApplicationTests для соответствия названию проекта;
- создан CarerController с эндпоинтами для добавления и получения опекунов;
- создан собственный класс исключений, которое выбрасывается при отсутствии опекуна в БД;
- в CarerService добавлен логгер, изменен метод по добавлению опекуна и добавлен метод по нахождению опекуна в БД;
- в pom.xml добавлены зависимости по подключению swagger, по валидации spring boot, по определению драйвера для БД;
- добавлено изображение для отправки в качестве схемы проезда в приют;
- в TelegramBotConfiguration при создании телеграм бота добавлена функция удаления старого списка команд перед добавлением нового;
- в TelegramBotUpdatesListener изменена ссылка на изображение схемы проезда в приют.

Garik-create committed:
Выполнена обработка нажатия кнопок с выходом в подменю этапа 2,3, Выполнена обработка команд меню с переходом на подменю этапа 2,3. В самих меню выполнен вывод заменяющего текста.

Garik-create committed:
Merge pull request #2 from Zhoss/1-feature-igor

Выполнена обработка нажатия кнопок с выходом в подменю этапа 2,3, Вып…

KlochkoVitaliy committed:
Добавлена документация по API

KlochkoVitaliy committed:
Merge pull request #3 from Zhoss/1-feature-Vitaliy

Добавлена документация по API

dmitry-bizin commented on pom.xml:
```diff
@@ -27,6 +27,51 @@
+
+        <dependency>
+            <groupId>jakarta.persistence</groupId>
+            <artifactId>jakarta.persistence-api</artifactId>
```
эта зависимость не нужна, она есть в spring-boot-starter-data-jpa

ZhosanDenis replied:
Почему то без этой зависимости не добавляются аннотации для сущностей

dmitry-bizin replied:
всё должно добавляться, аннотации должны быть из пакета jakarta.persistence

dmitry-bizin commented on pom.xml:
```diff
@@ -27,6 +27,51 @@
+
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
```
эта зависимость тоже не нужна

ZhosanDenis replied:
Тоже что и с jakarta - аннотация @PostConstruct (вроде она) становится красной, т.е. не импортируется.

dmitry-bizin replied:
всё должно добавляться, аннотации должны быть из пакета jakarta.persistence

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java:
```diff
@@ -0,0 +1,71 @@
+                    @PostMapping
+            public ResponseEntity<Carer>addCarer(@Parameter(required = true, allowEmptyValue = true, name = "ФИО Опекуна") @RequestParam String fullName,
+            @Parameter(required = true, allowEmptyValue = true, name = "Возраст опекуна")@RequestParam int age,
+            @Parameter(required = true, allowEmptyValue = true, example = "+7 999 5553399", name = "Номер телефона опекуна")@RequestParam String phoneNumber) {
```
а почему не в теле параметры передаются?

dmitry-bizin replied:
лучше бы в теле передавать

dmitry-bizin commented on src/main/resources/liquibase/scripts/scripts.sql:
```diff
@@ -0,0 +1,70 @@
+-- liquibase formatted sql
+
+-- changeset dzhosan:1
+CREATE TABLE dogs
```
на схеме и в сущностях почему-то нет этой таблицы

ZhosanDenis replied:
В следующем changeset эти таблицы дропнули, т.к. решили, что они не нужны. Будем снова создавать

dmitry-bizin commented on src/main/resources/liquibase/scripts/scripts.sql:
```diff
@@ -0,0 +1,70 @@
+    is_taken   BOOLEAN     NOT NULL DEFAULT FALSE
+);
+
+CREATE TABLE agreements
```
на схеме и в сущностях почему-то нет этой таблицы

Garik-create committed:
Убрал не используемый импорт. Убран не нужные зависимости.

Garik-create committed:
Убрал не используемый импорт. Убран не нужные зависимости. Частично заполнил JavaDoc. Код запускается, бот и Сваггер тоже.

ZhosanDenis committed:
Merge pull request #5 from Zhoss/1-feature-igor

Убрал не используемый импорт. Убран не нужные зависимости.

Garik-create committed:
Поправил README.md. Вставил свой chatId.

ZhosanDenis committed:
Изменения №1 от 08.04.2023
- добавлены таблицы в БД: dogs, agreements. Добавлены поля в таблицу carers: dog_id, agreement_id, birth_year. Удалены поля в таблице carers: age.
- Добавлены классы: Dog, DogRecord, DogRepository, DogService, DogController, DogNotFoundException, CarerRecord, Agreement;
- в класс Carer добавлены поля: birthYear(заместо age), OneToOne поля dog, agreement, dailyReport;
- в CarerController добавлены эндпоинты по редактированию и удалению опекунов;
- в CarerService добавлены методы по добавлению опекунов через CarerRecord, по редактированию и удалению опекунов;
- в DailyReport добавлены OneToOne поля carer, dog;
- проведен рефакторинг кода в TelegramBotUpdatesListener - оптимизировано количество условных операторов и использования методов.

Garik-create committed:
Добавил контроллер, сервис, репозиторий для DailyReport. Добавил эндпойнты поиска отчёта по ID опекуна и по ID опекуна и дате отчёта.

Garik-create committed:
Merge pull request #6 from Zhoss/1-feature-denis

Изменения №1 от 08.04.2023

Garik-create committed:
Частично добавил API к контроллеру DailyReportController.

Garik-create committed:
Merge branch 'dev' into 1-feature-igor

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java

Garik-create committed:
Merge pull request #7 from Zhoss/1-feature-igor

1 feature igor

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -0,0 +1,68 @@
+
+    private final DailyReportService dailyReportService;
+
+    private static final Logger logger = LoggerFactory.getLogger(DailyReportController.class);
```
LOGGER, т.к. static final

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -0,0 +1,68 @@
+            )
+    })
+
+    @GetMapping("/carer")
```
/carer/reports/carer ? не слишком много carer?))

Garik-create commented:
The first merge into master.
